sql/table.go

```diff
@@ -202,6 +205,13 @@ func encodeTableKey(b []byte, val parser.Datum) ([]byte, error) {
 		return encoding.EncodeFloat(b, float64(t)), nil
 	case parser.DString:
 		return encoding.EncodeString(b, string(t)), nil
+	case parser.DDate:
+		sec := t.Time.Unix()
+		return encoding.EncodeVarint(b, sec), nil
```

Comment:
No need for the variable here: `return encoding.EncodeVarint(b, t.Time.Unix()), nil`.